doc/nrf/ug_app_dev.rst

```diff
@@ -56,3 +56,46 @@ In order to define your own board, you can use the following Zephyr guides as re
 
 * :ref:`custom_board_definition` is a guide to adding your own custom board to the Zephyr build system.
 * :ref:`board_porting_guide` is a complete guide to porting Zephyr to your own board.
+
+Memory footprint optimization
+*****************************
+
+When developing an application, ROM and RAM footprint are important factors, especially when the firmware runs on the most resource-constrained devices e.g. nRF52810 or nRF52811.
+If you want your application to use the minimum required resources, follow the steps below and tune the |NCS| configuration parameters reducing the memory footprint:
+
+# Follow the basic hints for Zephyr :ref:`zephyr:optimizations`
```

Comment:
We could also point to this sample in upstream: https://docs.zephyrproject.org/2.3.0/samples/basic/minimal/

Reply:
Good point - added